### PR TITLE
Bug 1584321 - Handle static client scopes internally

### PR DESCRIPTION
**Pull request by djmitche**
With this change, each service declares its scopes in
`services/<service>/scopes.yml`, and that data is gathered during
generation and placed in a convenient place for the auth service.  The
Auth service then interpolates those scopes into the configured
STATIC_CLIENTS, preserving the supplied accessToken and applying the
service's configured azure accountId.

This is messier than it should be, but until we can find some time to focus on software development, it will have to do.

Bugzilla Bug: [1584321](https://bugzilla.mozilla.org/show_bug.cgi?id=1584321)

### CODE REVIEW

**Comment by djmitche** [comment]
Note that this also removes the unnecessary scope `auth:azure-table-access:${azureAccountId}/Secrets` for the secrets service.

File: services/auth/src/data.js [hunk 1]
```diff
@@ -112,20 +113,61 @@ Client.prototype.json = function(resolver) {
...
+      assert(!client.scopes, 'scopes are not allowed in configuration for static/taskcluster clients');
+    } else {
+      assert(client.scopes instanceof Array, 'expected scopes to be an array of strings');
+      assert(typeof client.description === 'string', 'expected description to be a string');
+      assert(client.scopes.every(s => typeof s === 'string'), 'scopes must be strings');
+    }
+  }
+
+  // check that we have all of the expected static/taskcluster clients, and no more.  The staticClients
+  // are generated from `services/*/scopes.yml` for all of the other services.
```
Comment (imbstack) [inline services/auth/src/data.js]: this makes me feel a lot safer about this

File: infrastructure/tooling/src/generate/generators/k8s.js [hunk 2]
```diff
@@ -89,7 +88,7 @@ const renderTemplates = async (name, vars, procs, templates) => {
         const file = `taskcluster-${name}-service-${proc}.yaml`;
         ingresses.push({
           projectName: `taskcluster-${name}`,
-          paths: conf['paths'] || [`/api/${name}/*`], // TODO: This version of config is only for gcp ingress :(
+          paths: conf['paths'] || ['/api/' + name + '/*'], // TODO: This version of config is only for gcp ingress :(
```
[the remark this answers is not in the document]
Comment (djmitche) [reply infrastructure/tooling/src/generate/generators/k8s.js]: Oh, I meant to remove that -- `vim` doesn't understand backticks for some reason, and the `/*` makes it think the rest of the file is a comment.  I can remove.

File: infrastructure/tooling/src/dev/helm.js [hunk 3]
```diff
@@ -54,19 +69,27 @@ const actions = [
...
+        return {
+          'target-templates': await execCommand({
+            command,
+            dir: REPO_ROOT,
+            keepAllOutput: true,
+            utils,
+          }),
+        };
+      } finally {
+        tmpfile.unlink();
```
[the remark this answers is not in the document]
Comment (djmitche) [reply infrastructure/tooling/src/dev/helm.js]: Nope, good catch!

**Pull request merged**